### DISCUSSION TITLE
Amazon Silk Browser Support

### DISCUSSION THREAD

**Discussion by mstackler** [jgyates/genmon#1059] [Ideas]
Amazon Silk browser is included with all of the echo home products.  It currently times out and returns to main menu after 10 minutes.  I have an echo show 15 that i am attempting to use as a generator monitor display however it times out after 10 minutes.  I have looked around and this is a common complaint with this browser.  Someone created a script that you can add to the webpage that runs something in the background tricking the browser into staying open.  can something like this be added to the add on page?

Here is a link to what they did
https://gitlab.com/DaGammla/keep-silk-open

**Comment by jgyates** [comment]
I will look into this. I would not want to just add the reference to the code and have it pull it from the internet for both security concerns and I would not want genmon to rely on the internet to check the status of the generator. If I implimnet this i would copy the needed script into the genmon tree so it would be available on the local network. I will update this thread when I have something to report.

**Comment by jgyates** [comment]
@mstackler 

i added the code reference above, but I only added it to this URL

http://192.168.1.10:8000/index_verbose.html

Use the index_verbose.html page to make this code be called. I tried to test this the best I could. I was able to validate the code gets called, however I only have a Fire TV box with the Silk browser app. Once the screen saver kicks in it takes you back to the home page even with this code added. Let me know how your testing goes on the echo tablet after you update from the about page.
